cmd/server/api/router_test.go

```diff
@@ -0,0 +1,69 @@
+package api_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/freeverseio/laos-universal-node/cmd/server/api"
+	"github.com/freeverseio/laos-universal-node/cmd/server/api/mock"
+	stateMock "github.com/freeverseio/laos-universal-node/internal/platform/state/mock"
+	"github.com/gorilla/mux"
+	"go.uber.org/mock/gomock"
+)
+
+func TestCORS(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockRPCHandler := mock.NewMockRPCHandler(mockCtrl)
+	state := stateMock.NewMockService(mockCtrl)
+
+	mockRPCHandler.EXPECT().PostRPCRequestHandler(gomock.Any(), gomock.Any()).Times(1)
+	mockRPCHandler.EXPECT().SetStateService(gomock.Any()).Times(1)
+	router := mux.NewRouter()
+	api.Routes(mockRPCHandler, router, state)
+
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	client := ts.Client()
+
+	tests := []struct {
+		method       string
+		url          string
+		status       int
+		allowOrigin  string
+		allowMethods string
+	}{
+		{"POST", "/", http.StatusOK, "*", "POST, OPTIONS"},
+		{"OPTIONS", "/", http.StatusOK, "*", "POST, OPTIONS"},
+		{"GET", "/", http.StatusMethodNotAllowed, "", ""},
+	}
+
+	for _, tc := range tests {
+		req, err := http.NewRequest(tc.method, ts.URL+tc.url, http.NoBody)
+		if err != nil {
+			t.Errorf("could not create request: %v", err)
+		}
+		res, err := client.Do(req)
+		if err != nil {
+			t.Errorf("could not send request: %v", err)
```

Comment:
If `res` is `nil`, the next check will panic.

Either you replace all `t.Errorf` calls for `t.Fatalf` or you check that `res` is not `nil` on every `if`